File: app/src/main/java/com/konopko/stocktest/ActivityBase.java
package com.konopko.stocktest;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

public abstract class ActivityBase<T extends ViewDataBinding> extends AppCompatActivity {

    protected T binding;
    protected abstract int getLayoutId();
    protected abstract String getPageTitle();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, getLayoutId());

        if (getSupportActionBar() != null)
            getSupportActionBar().setTitle(getPageTitle());
    }
}

File: app/src/main/java/com/konopko/stocktest/ActivityBase.kt
package com.konopko.stocktest

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

abstract class ActivityBase<T : ViewDataBinding> : AppCompatActivity() {

    protected lateinit var binding: T
    protected abstract fun getLayoutId(): Int
    protected abstract fun getPageTitle(): String?

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, getLayoutId())

        supportActionBar?.title = getPageTitle()
    }
}